sec_certs_page/common/bindings.py

```diff
@@ -0,0 +1,510 @@
+    """
+    parsed_url = urllib.parse.urlparse(url)
+    path = urllib.parse.unquote(parsed_url.path.strip("/"))
+    return path
+
+
+def github_url_to_api(url: str) -> str:
+    """ Transform the standard public github URL to a github API URL
+
+    Args:
+        url (str): URL to transform
+
+    Returns:
+        str: Transformed API URL
+    """
+    parsed_url = urllib.parse.urlparse(url)
+    api_url = f"https://api.github.com/repos{parsed_url.path}"
+    if parsed_url.query:
+        api_url += f"?{parsed_url.query}"
+    return api_url
+
+
+def get_filename_from_url(url: str) -> str:
+    """ Parse out the filename from a given url
+
+    Args:
+        url (str): URL to parse
+
+    Returns:
+        str: Parsed out filename
+    """
+    path_components = url.split("/")
+    filename = path_components[-1]
+    if not filename:
+        filename = path_components[-2]
+    return os.path.basename(filename)
+
+
+def verify_timestamp(timestamp: str) -> bool:
+    """ Verifies timestamp validity
+
+    Args:
+        timestamp (str): timestamp to validate
+
+    Returns:
+        bool: Whether the given timestamp is valid
+    """
+    try:
+        datetime.datetime.fromisoformat(timestamp)
+        return True
+    except ValueError:
+        return False
+
+
+def verify_url(url: str) -> bool:
+    """ Verifies URL validity
+
+    Args:
+        url (str): URL to validate
+
+    Returns:
+        bool: Whether the given URL is valid
+    """
+    try:
+        result = urllib.parse.urlparse(url)
+        return all([result.scheme, result.netloc])
+    except ValueError:
+        return False
+
+
+def verify_jwt(data: dict[str, Union[str, object]]) -> bool:
+    """ Verifies the given object jwt validity
+
+    Args:
+        data (dict[str, Union[str, object]]): object to validate
+
+    Returns:
+        bool: Whether the JWT in the data object is valid
+    """
+    jwt_token = data['JWT']
+    del data["JWT"]
+    encoded_jwt = jwt.encode(
+        data, key=current_app.config["BINDINGS_PUBLIC_KEY"], algorithm="RS256")
+    if not jwt_token:
+        click.echo("JWT token is missing")
+        return False
+    if encoded_jwt != jwt_token:
+        return False
+    return True
+
+
+async def download_file(file_url: str, file_path: str, sha256: str = "", verbose: bool = False) -> None:
+    """ Download a file from the given URL
+
+    Args:
+        file_url (str): URL of the file to download
+        file_path (str): Path to where to download the file
+        sha256 (str, optional): Expected hash of the downloaded file. Defaults to "".
+        verbose (bool, optional): Whether to click.echo out extra information. Defaults to False.
+    """
+    if verbose:
+        click.echo(f"Downloading {file_url}...")
+    response = requests.get(file_url, timeout=1000)
+    if sha256:
+        file_sha256 = hashlib.sha256(response.content).digest().hex()
+        if file_sha256 != sha256:
+            click.echo(
```

Comment:
This should abort right? Why is there verification at all if it still saves the file and continues?